tcms/rpc/tests/test_testcase.py

```diff
@@ -16,13 +21,66 @@
                                   TestPlanFactory, UserFactory, VersionFactory)
 
 
-class TestNotificationRemoveCC(APITestCase):
-    """ Tests the XML-RPC testcase.notication_remove_cc method """
+class TestValidateEmail(unittest.TestCase):
+    def test_non_list_email_collection(self):
+        with self.assertRaisesRegex(TypeError, 'cc_list should be a list object.'):
+            _validate_cc_list(('example@example.com',))
+
+    def test_invalid_email(self):
+        with self.assertRaises(ValidationError):
+            _validate_cc_list(['@example.com'])
+
+    def test_valid_email_list(self):
+        _validate_cc_list(['example@example.com'])
+
+
+class TestAddNotificationCCPermission(APIPermissionsTestCase):
+    permission_label = 'testcases.change_testcase'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.default_cc = 'example@example.com'
+        self.testcase = TestCaseFactory()
+
+    def verify_api_with_permission(self):
+        self.rpc_client.TestCase.add_notification_cc(self.testcase.pk, [self.default_cc])
+        self.assertEqual(len(self.testcase.emailing.get_cc_list()), 1)
+        self.assertEqual(self.testcase.emailing.get_cc_list(), [self.default_cc])
+
+    def verify_api_without_permission(self):
+        with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
+            self.rpc_client.TestCase.add_notification_cc(self.testcase.pk, [self.default_cc])
+
+
+class TestGetNotificationCCPermission(APIPermissionsTestCase):
+    """ Tests the XML-RPC TestCase.get_notification_cc permissions """
+
+    permission_label = 'testcases.view_testcase'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.default_cc = 'example@example.com'
+        self.testcase = TestCaseFactory()
+        self.testcase.emailing.add_cc(self.default_cc)
+
+    def verify_api_with_permission(self):
+        cc_list = self.rpc_client.TestCase.get_notification_cc(self.testcase.pk)
+        self.assertListEqual(cc_list, [self.default_cc])
+
+    def verify_api_without_permission(self):
+        with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
+            self.rpc_client.TestCase.get_notification_cc(self.testcase.pk)
+
+
+class TestRemoveNotificationCC(APITestCase):
+    """ Tests the XML-RPC TestCase.remove_notification_cc method """
 
     def _fixture_setup(self):
         super()._fixture_setup()
 
-        self.default_cc = 'example@MrSenko.com'
+        self.default_cc = 'example@example.com'
```

Comment:
Missing scenario with invalid TC ID which asserts we get a DoesNotExist exception. See above.